ui/components/MesherySMIResults.js

```diff
@@ -0,0 +1,217 @@
+          const data = smi_results.results[rowMeta.dataIndex].more_details.map((val) => {
+            return [val.smi_specification,val.assertions,val.time,"alpha1/v1",val.capability,val.status,val.reason] 
+          })
+          const colSpan = rowData.length + 1
+          return (
+            <TableRow>
+              <TableCell colSpan={colSpan}>
+                <Paper elevation={4} >
+                  <Table aria-label="a dense table">
+                    <TableHead>
+                      <TableRow>
+                        {column.map((val) => (<TableCell colSpan={colSpan}>{val}</TableCell>))}
+                      </TableRow>
+                    </TableHead>
+                    <TableBody>
+                      {data.map((row) => (
+                        <TableRow >
+                          {row.map(val => (<TableCell colSpan={colSpan}>{val}</TableCell>))}
+                        </TableRow>
+                      ))}
+                    </TableBody>
+                  </Table>
+                </Paper>
+              </TableCell>
+            </TableRow>
+          );
+        },
+        onTableChange: (action, tableState) => {
+          const sortInfo = tableState.announceText? tableState.announceText.split(' : '):[];
+          let order='';
+          if(tableState.activeColumn){
+            order = `${columns[tableState.activeColumn].name} desc`;
+          }
+
+          switch (action) {
+            case 'changePage':
+              self.fetchSMIResults(tableState.page, self.state.smi_pageSize, self.state.smi_search, self.state.smi_sortOrder);
+              break;
+            case 'changeRowsPerPage':
+              self.fetchSMIResults(self.state.smi_page, tableState.rowsPerPage, self.state.smi_search, self.state.smi_sortOrder);
+              break;
+            case 'search':
+              if (self.searchTimeout) {
+                clearTimeout(self.searchTimeout);
+              }
+              self.searchTimeout = setTimeout(() => {
+                if (self.state.search !== tableState.searchText) {
+                  self.fetchSMIResults(self.state.smi_page, self.state.smi_pageSize, tableState.searchText !== null ? tableState.searchText : '', self.state.smi_sortOrder);
+                }
+              }, 500);
+              break;
+            case 'sort':
+              if (sortInfo.length == 2) {
+                if (sortInfo[1] === 'ascending') {
+                  order = `${columns[tableState.activeColumn].name} asc`;
+                } else {
+                  order = `${columns[tableState.activeColumn].name} desc`;
+                }
+              }
+              if (order !== this.state.sortOrder) {
+                self.fetchSMIResults(self.state.smi_page, self.state.smi_pageSize, self.state.smi_search, order);
+              }
+              break;
+          }
+        },
+      }
+
+      return (
+        <NoSsr>
```

Comment:
Is there any need for <NoSsr> wrappers here?

Reply:
Was just following the design scheme of other pages. But yes, I think we wont need it in this case since we are not using the window API or other client side dependencies anywhere.

Reply:
We can create an issue for the style guide in react components, I dont guess each mui component needs NoSsr except timer utility in performance